### ISSUE TITLE
python 3.6 => move to at least 3.7?

### ISSUE THREAD

**Issue by darked89** [maciosz/HERON#2]
Hello,

I have tried to install HERON using either a conda environment with python 3.6 or compiling python 3.6.15 from source. 
So far I have not succeeded. 

My suspicion is that since the python 3.6 is no longer maintained therefore various python libs may not install correctly.   pysam will not install with pip from freshly compiled  python 3.6.15. Upgrading pip "fixes" it. so pysam is reported as installed, but then I get a lot of:
```
Link requires a different Python (3.6.15 not in: '>=3.7')
```

The next step:
```
pip3 install sklearn  --verbose

Collecting sklearn
  Downloading sklearn-0.0.post1.tar.gz (3.6 kB)

ERROR: Could not find a version that satisfies the requirement sklearn (from versions: 0.0, 0.0.post1, 0.0.post2)
ERROR: No matching distribution found for sklearn
```

In short: would it be possible to update HERON so it runs using python 3.7 (or newer)?

An alternative would be to package it inside a container.

Many thanks for your help

Darek Kedra

**Comment by maciosz** [comment]
Hi Darek,

Thanks for bringing this up. I will look into it during this week and get back to you.

**Comment by darked89** [comment]
Thanks for a quick reply. 

The bad thing is that  endo of support for Python 3.7 will be: 2023-06-27. 
On the other hand going beyond 3.7 will require fixing some Cython(?) code. I have seen +/- same error trying to start HERON with python 3.9:

https://stackoverflow.com/questions/69591726/after-python-to-c-conversion-via-cython-getting-undefined-symbol-py-checkrecur

Hope it helps.

DK
